src/ImageCarouselReact/widget/ImageCarouselReact.tsx

```diff
@@ -0,0 +1,355 @@
+     * move contructor work into widget prototype at bottom of the page. 
+     */
+    constructor(args?: Object, elem?: HTMLElement) {
+        // Do not add any default value here... it wil not run in dojo!     
+        super() ;
+        return new dojoImageCarouselReact(args, elem);
+    }
+    /**
+     * Create properities for react component based on modeler configurations
+     */
+    public createProps(): ImageCarouselProps {
+        let contextId = this.contextObj ? this.contextObj.getGuid() : "";
+        return {
+            captionAttr: this.captionAttr,
+            contextId,
+            controls: this.controls,
+            data: this.data,
+            dataSourceMicroflow: this.dataSourceMicroflow,
+            descriptionAttr: this.descriptionAttr,
+            entityConstraint: this.entityConstraint,
+            height: this.height,
+            heightUnits: this.heightUnits,
+            imageClickMicroflow: this.imageClickMicroflow,
+            imageEntity: this.imageEntity,
+            imageSource: this.imageSource,
+            indicators: this.indicators,
+            interval: this.interval,
+            isLoading: this.isLoading,
+            onClickEvent: this.onClickEvent,
+            openPage: this.openPage,
+            // openPageModal: this.location, // TODO add to interface later
+            pauseOnHover: this.pauseOnHover,
+            requiresContext: this.requiresContext,
+            slide: this.slide,
+            staticImageCollection: this.staticImageCollection,
+            width: this.width,
+            widthUnits: this.widthUnits,
+        };
+    }
+    /**
+     * dijit._WidgetBase.postCreate is called after constructing the widget.
+     * React widget component, passing in all the props
+     */
+    public postCreate() {
+        logger.debug(this.id + ".postCreate");
+        this.updateRendering();
+    }
+    /**
+     * mxui.widget._WidgetBase.update is called when context is changed or initialized. 
+     *
+     */
+    public update(obj: mendix.lib.MxObject, callback?: Function) {
+        logger.debug(this.id + ".update");
+        this.contextObj = obj;
+        this.updateData(() => {
+            this.isLoading = false;
+            this.updateRendering(callback);
+        });
+        this._resetSubscriptions();
+    }
+    /**
+     * mxui.widget._WidgetBase.uninitialize is called when the widget is destroyed.
+     * will need to unmount react components
+     */
+    public uninitialize() {
+        logger.debug(this.id + ".uninitialize");
+        // Clean up listeners, helper objects, etc. 
+        // There is no need to remove listeners added with this.connect 
+        this._unsubscribe();
+        ReactDOM.unmountComponentAtNode(this.domNode);
+    }
+    /**
+     *  called to render the interface 
+     */
+    private updateRendering(callback?: Function) {
+        logger.debug(this.id + ".updateRendering");
+        ReactDOM.render(
+            <ImageCarousel
```

Comment:
@akileng56 All jsx is removed so this file can be rename to .ts
The components will stay .jsx